apps/prairielearn/src/ee/pages/institutionAdminAdmins/institutionAdminAdmins.ts

```diff
@@ -16,7 +35,80 @@ router.get(
       access_as_administrator: res.locals.access_as_administrator,
     });
 
-    res.send(InstitutionAdminAdmins({ institution, resLocals: res.locals }));
+    const rows = await queryValidatedRows(
+      sql.select_admins,
+      { institution_id: institution.id },
+      InstitutionAdminAdminsRowSchema,
+    );
+
+    res.send(
+      InstitutionAdminAdmins({
+        institution,
+        rows,
+        uidsLimit: MAX_UIDS,
+        resLocals: res.locals,
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const institution = await selectAndAuthzInstitutionAsAdmin({
+      institution_id: req.params.institution_id,
+      user_id: res.locals.authn_user.user_id,
+      access_as_administrator: res.locals.access_as_administrator,
+    });
+
+    if (req.body.__action === 'addAdmins') {
+      const uids = parseUidsString(req.body.uids, MAX_UIDS);
+      const validUids: string[] = [];
+      const invalidUids: string[] = [];
+      await runInTransactionAsync(async () => {
+        for (const uid of uids) {
+          const user = await selectUserByUid(uid);
+
+          // Specifically check that the user is in the institution to prevent
+          // someone from enumerating users in other institutions.
```

Comment:
I feel like the right way to handle this is via invites, so that we don't need to place restrictions on user-in-same-institution while still allowing full privacy. I'm ok with not implementing invites right now if you don't want to do it (PT institution admins do use invites, fwiw).

Reply:
I'm going to skip building an invite system for now. While I do think we should do that eventually, that can probably wait until we build an invite system in more valuable places (enrollments, course staff, etc).